### ISSUE TITLE
Dask Dataframe date arithmetic gives 1+ 1 = 3

### ISSUE THREAD

**Issue by nils-braun** [dask/dask#7624]
**What happened**:

When adding a `pd.tseries.offsets.DateOffset(months=12);` to a dask dataframe containing a `datetime`, the addition seems to be done twice (?) - but only when using a distributed cluster.

It gives correct results when (a) using years instead of months in `DateOffset` (b) not using a dask.distributed cluster, (c) using a number of months not dividable by 12 and (d) with a newer pandas version (1.2.4).

Even though a newer pandas version seems to fix this, I thought it is good to report (if not know already), because this seems to be a really strange bug to me :-/

**Minimal Complete Verifiable Example**:

```python
import dask.dataframe as dd
from dask.distributed import Client
import pandas as pd
from datetime import datetime

# (b) Works correctly without this line
client = Client()

date = datetime(2021, 10, 3, 15, 53, 42, 47)
df = dd.from_pandas(pd.DataFrame({"d": [date]}), npartitions=1)
# (a) works correctly with DateOffset(years=1)  or (c) DateOffset(months=13)
# but also try DateOffset(months=24) or DateOffset(months=36) (all wrong)
delta = pd.tseries.offsets.DateOffset(months=12)

print((df + delta).compute())  # Gives 2023, instead of 2022.
```
**Environment**:

- Dask version: 2021.4.1
- Pandas: 1.1.5
- Python version: 3.8.8
- Operating System: Ubuntu
- Install method (conda, pip, source): Conda

**Comment by jsignell** [comment]
Oh weird. Thanks for raising this @nils-braun. I can reproduce on the oldest pandas version that dask supports - 0.25.0. Do you know if this is a regression?

**Comment by jsignell** [comment]
Ok I am pretty sure it is this: https://github.com/pandas-dev/pandas/issues/34511 which makes sense because it was resolved in pandas 1.2. I suppose we could raise a warning for this case, but it seems equally valid to do nothing and let people upgrade their pandas.

**Comment by nils-braun** [comment]
Thanks @jsignell! (I should have searched the pandas issues better). That looks exactly like this issue. I am also fine with keeping it like it is, as we now understand the root cause.
